metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -305,11 +307,8 @@ final class BuildTargets(
         names match {
           case Directories.dependenciesName :: jarName :: _ =>
             // match build target by source jar name
-            inverseDependencySources
```

Comment:
In term of memory this is nothing because a build often have less than 100 jar files. Even if it contains 10.000 jar files the Map is still very small.

I needed to create the `sourceJarFile` method for the `SourcePathAdapter` and this is the simplest implementation I could think of.